src/BaGet.Core/Search/DatabaseSearchService.cs

```diff
@@ -106,15 +106,38 @@ public DatabaseSearchService(IContext context, IFrameworkCompatibilityService fr
             };
         }
 
-        public Task<AutocompleteResponse> ListPackageVersionsAssync(
-            string packageId,
-            bool includePrerelease,
-            bool includeSemVer2,
+        public async Task<AutocompleteResponse> ListPackageVersionsAsync(
+            AutocompleteRequest request,
             CancellationToken cancellationToken)
         {
-            // TODO: Support versions autocomplete.
-            // See: https://github.com/loic-sharma/BaGet/issues/291
-            throw new NotImplementedException();
+            IQueryable<Package> search = _context.Packages;
+
+            if (!string.IsNullOrEmpty(request.Query))
+            {
+                var query = request.Query.ToLower();
+                search = search.Where(p => p.Id.ToLower().Equals(query));
+            }
+
+            search = AddSearchFilters(
+                search,
+                request.IncludePrerelease,
+                request.IncludeSemVer2,
+                request.PackageType,
+                frameworks: null);
+
+            var results = await search
+                .Distinct()
+                .Select(p => p.Version.ToString())
+                .ToListAsync(cancellationToken);
+
+            results.Sort();
```

Comment:
The results don't need to be sorted, so let's remove this line. (Also, string sorting isn't quite right here as it yields a different order than semver sorting. To get the desired result we'd have to parse all the versions before sorting them)